ebean-migration/src/main/java/io/ebean/migration/runner/LocalMigrationResources.java

```diff
@@ -45,9 +47,60 @@ boolean readInitResources() {
    * Read all the migration resources (SQL scripts) returning true if there are versions.
    */
   boolean readResources() {
+    if (readFromIndex()) {
+      return true;
+    }
     return readResourcesForPath(migrationConfig.getMigrationPath());
   }
 
+  private boolean readFromIndex() {
+    final var base = "/" + migrationConfig.getMigrationPath() + "/";
+    final var basePlatform = migrationConfig.getBasePlatform();
+    final var indexName = "idx_" + basePlatform + ".migrations";
+    URL idx = resource(base + indexName);
+    if (idx != null) {
+      return loadFromIndexFile(idx, base);
+    }
+    idx = resource(base + basePlatform + '/' + indexName);
+    if (idx != null) {
+      return loadFromIndexFile(idx, base + basePlatform + '/');
+    }
+    final var platform = migrationConfig.getPlatform();
+    idx = resource(base + platform + indexName);
+    if (idx != null) {
+      return loadFromIndexFile(idx, base + platform + '/');
+    }
+    return false;
+  }
+
+  private URL resource(String base) {
+    return LocalMigrationResources.class.getResource(base);
+  }
+
+  private boolean loadFromIndexFile(URL idx, String base) {
+    try (var reader = new LineNumberReader(new InputStreamReader(idx.openStream()))) {
+      String line;
+      while ((line = reader.readLine()) != null) {
+        if (!line.isEmpty()) {
+          final String[] pair = line.split(",");
+          if (pair.length == 2) {
+            final var checksum = Integer.parseInt(pair[0]);
```

Comment:
@rbygrave I' wondering, if this affects #91

We currently have some minor issues when copying a database from staging to testing, if the parameters in the script are different.

For example, we use the `${tablespace}` placeholder in the DB2 scripts. In our staging, this parameter is set to "STAGING", on our testings, this is set to "TESTING".

As Ebean currently computes the checksum as last step, we cannot simply copy the data from staging DB to a testing DB as the checksum will change, if the parameter value changes. I think using the checksum in the IDX file will break things here, if the script contains parameters.

Reply:
It breaks things _IF_ there are existing migrations like this that are already run and then using the index file is adopted (the checksums will be different yes).